index.js

```diff
@@ -244,10 +254,151 @@ class BitbucketScm extends Scm {
-    */
+     * Get a commit sha for a specific repo#branch
+     * @method getCommitSha
+     * @param  {Object}   config            Configuration
+     * @param  {String}   config.scmUri     The scmUri
+     * @param  {String}   config.token      The token used to authenticate to the SCM
+     * @return {Promise}                    Resolves to the sha for the scmUri
+     */
+    _getCommitSha(config) {
+        const scm = getScmUriParts(config.scmUri);
+        const branchUrl =
+            `${REPO_URL}/${scm.repoId}/refs/branches/${scm.branch}?access_key=${config.token}`;
+        const options = {
+            url: branchUrl,
+            method: 'GET'
+        };
+
+        return this.breaker.runCommand(options)
+            .then((response) => {
+                if (response.statusCode !== 200) {
+                    throw new Error(`STATUS CODE ${response.statusCode}: ${response.body}`);
+                }
+
+                return response.body.target.hash;
+            });
+    }
+
+    /**
+     * Fetch content of a file from Bitbucket
+     * @method getFile
+     * @param  {Object}   config              Configuration
+     * @param  {String}   config.scmUri       The scmUri
+     * @param  {String}   config.path         The file in the repo to fetch
+     * @param  {String}   config.token        The token used to authenticate to the SCM
+     * @param  {String}   config.ref          The reference to the SCM, either branch or sha
+     * @return {Promise}                      Resolves to the content of the file
+     */
+    _getFile(config) {
+        const scm = getScmUriParts(config.scmUri);
+        const fileUrl = `${API_URL_V1}/repositories/${scm.repoId}` +
+            `/src/${config.ref}/${config.path}?access_key=${config.token}`;
+        const options = {
+            url: fileUrl,
+            method: 'GET'
+        };
+
+        return this.breaker.runCommand(options)
+            .then((response) => {
+                if (response.statusCode !== 200) {
+                    throw new Error(`STATUS CODE ${response.statusCode}: ${response.body}`);
+                }
+
+                return response.body.data;
+            });
+    }
+
+    /**
+     * Get a user's permissions on a repository
+     * @method _getPermissions
+     * @param  {Object}   config            Configuration
+     * @param  {String}   config.scmUri     The scmUri
+     * @param  {String}   config.token      The token used to authenticate to the SCM
+     * @return {Promise}                    Resolves to permissions object with admin, push, pull
+     */
+    _getPermissions(config) {
+        const scm = getScmUriParts(config.scmUri);
+        const getPerm = (repoId, desiredAccess, token) => {
+            const [owner, uuid] = repoId.split('/');
+            const options = {
+                url: `${API_URL_V2}/repositories/${owner}`,
+                method: 'GET'
+            };
+
+            if (desiredAccess === 'admin') {
```

Comment:
I used switch case at first but that makes the code uglier since you need `break` everywhere